src/utilities/valueFromAST.js

```diff
@@ -111,6 +117,7 @@ export function valueFromAST(
   );
 
   const parsed = type.parseLiteral(valueAST);
+  // TODO: Should be this condition ommited?
```

Comment:
This is a little bit tricky now, but I think the right decision is to leave this check alone.

Right now the definition for parseLiteral is to either return a value, or return null (or some nullish value) if the parse failed. I don't think it's ever appropriate for parseLiteral to return null and have that be intended as a null value to be used.